### DISCUSSION TITLE
NSGA-II multi-objective problem

### DISCUSSION THREAD

**Discussion by MichaelSchlecht** [anyoptimization/pymoo#442] [Q&A]
Hey community!

I am using NSGAII for optimization in two multi-objective problems (3 and 8) objectives. I lately read in the list of algorithms (https://pymoo.org/algorithms/list.html) about NSGA-III: "An improvement of NSGA-II developed for multi-objective optimization problems with more than two objectives." In the description of NSGA-II i did not read about this limitation. Can somebody confirm that NSGA-II is able to do multi-objective with 3 and 8 objective functions and maybe provide the limitations?

Also, I was wondering about the number of non-dominated solution (n_nds) in the results. In my cases, it was always limited to 40. I think, especially with in creasing number of objectives, this number should increase. Is n_nds restricted by an editable parameter?

Thanks in advance.

**Comment by blankjul** [comment]
NSGA-II has been proposed mostly for 2 objectives. However, to get more solutions you have to increase `pop_size`. 

For more objectives, NSGA-III is the better choice. There, you will ideally find one non. dom. solution per reference-direction you provide.
